spec/models/timeline_spec.rb

```diff
@@ -0,0 +1,181 @@
+        it "returns an empty array" do
+          timeline = Timeline.new(events, [])
+          expect(timeline.events).to eq []
+        end
+      end
+
+      describe "with a TTA activity report" do
+        it "returns an array with only the TTA event" do
+          timeline = Timeline.new(events, [tta_activity_report])
+
+          expect(timeline.events.count).to be 1
+          expect(timeline.events.first).to be_a Timeline::TtaEvent
+          expect(timeline.events.first.name).to eql display_id
+        end
+      end
+
+      describe "multiple TTA activity reports" do
+        it "returns an array with TTA events in desc order" do
+          timeline = Timeline.new(events, [tta_activity_report, tta_activity_report_2])
+          first_report = timeline.events.first
+          second_report = timeline.events.last
+
+          expect(timeline.events.count).to be 2
+          expect(first_report.date > second_report.date).to be true
+        end
+      end
+    end
+
+    describe "with needed complaint events" do
+      let(:issue_last_updated) { 1.day.ago.strftime("%F") }
+      let(:creation_date) { 6.days.ago.strftime("%F") }
+      let(:initial_contact_date) { 1.week.ago.strftime("%F") }
+
+      let(:events) do
+        {
+          issueLastUpdated: issue_last_updated,
+          creationDate: creation_date,
+          initialContactDate: initial_contact_date,
+          otherRandomAttribute: "test attribute"
+        }
+      end
+
+      describe "with no TTA activity report" do
+        it "returns an array with just ComplaintEvent objects" do
+          timeline = Timeline.new(events, [])
+
+          expect(timeline.events.count).to be 3
+          timeline.events.each do |event|
+            expect(event).to be_a Timeline::ComplaintEvent
+          end
+        end
+
+        it "returns an array with objects in descending order" do
+          timeline = Timeline.new(events, [])
+          expect(timeline.events.first.date).to eq Date.parse(events[:issueLastUpdated])
+          expect(timeline.events.second.date).to eq Date.parse(events[:creationDate])
+          expect(timeline.events.last.date).to eq Date.parse(events[:initialContactDate])
+        end
+      end
+
+      describe "with a TTA activity report" do
+        it "returns an array with all the necessary events" do
+          timeline = Timeline.new(events, [tta_activity_report])
+
+          expect(timeline.events.count).to be 4
+        end
+      end
+    end
+  end
+
+  describe "ComplaintEvent" do
+    let(:date) { 1.day.ago.strftime("%F") }
+    let(:name) { "issueLastUpdated" }
+    let(:event_param) { [name, date] }
+
+    describe "#init" do
+      it "assigns attr_readers" do
+        event = Timeline::ComplaintEvent.new(event_param)
+        expect(event.name).to eq name
+        expect(event.date).to eq Date.parse(date)
+      end
+    end
+
+    describe "#label" do
+      it "returns the correct label" do
```

Comment:
[suggestion] I think testing the logic is way more important than testing all of the key/value pairs. I'd prefer to see one test that grabs the value out of the hash and one test that uses the fallback.

This test is a bit of an issue because it'll pass but we don't know if it's because of the hash or the fallback.